x-pack/plugins/cross_cluster_replication/public/register_routes.js

```diff
@@ -12,19 +12,25 @@ import { BASE_PATH } from '../common/constants/base_path';
 import { renderReact } from './app';
 import { setHttpClient } from './app/services/api';
 
+let elem;
+
 const CCR_REACT_ROOT = 'ccrReactRoot';
 
-routes.when(`${BASE_PATH}/:view?/:id?`, {
+const unmountReactApp = () => elem && unmountComponentAtNode(elem);
+
+routes.when(`${BASE_PATH}/:section?/:view?/:id?`, {
   template: template,
   controllerAs: 'ccr',
   controller: class CrossClusterReplicationController {
     constructor($scope, $route, $http) {
+      unmountReactApp();
```

Comment:
Could we add a comment explaining why we need to do this?

```js
// React-router's <Redirect> will cause this controller to re-execute without the $destroy
// handler being called. This means the app will re-mount, so we need to unmount it first
// here.
```

Reply:
Totally right, and right after I'll send an email to all kibana devs as a warning.